auth/oidc.go

```diff
@@ -82,12 +101,46 @@ type OIDCOptions struct {
 // with an OpenID Connect provider.
 type OIDCClient struct {
 
+	// configLock is a reader/writer mutual exclusion lock on config.
+	configLock sync.RWMutex
```

Comment:
Is it ever the case that we need to update the config and the verifier independently?  It looks like they are both associated with an update to the provider config?  If that's the case, using a single mutex would make this cleaner.